File: src/main/java/com/edu/uniquindio/ruralstay/dto/CasaDetallDTO.java
package com.edu.uniquindio.ruralstay.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class CasaDetallDTO {
    private Long codigo;
    private Integer plazasParqueo;
    private String descripcion;
    private PropietarioSimpleDTO propietario;
    private List<CocinaDTO> cocinas;
    private List<HabitacionDTO> habitaciones;
    private List<BañoDTO> banos;
}

File: frontend-angular/src/app/models/casa-detall-dto.ts
import { CocinaDto } from './cocina-dto';
import { HabitacionDto } from './habitacion-dto';
import { PropietarioSimpleDto } from './propietario-simple-dto';
import { BanoDto } from './baño-dto';

export interface CasaDetallDto {
  codigo: number;
  plazasParqueo: number;
  descripcion: string;
  propietario: PropietarioSimpleDto;
  cocinas: CocinaDto[];
  habitaciones: HabitacionDto[];
  banos: BanoDto[];
  fotos?: string[];
}
